axlearn/common/attention.py

```diff
@@ -2771,11 +2788,14 @@ def _linear2(x):
 
         remat_pt2 = "linear2"
         if cfg.structure == "prenorm":
+            inputs = maybe_shard(inputs, cfg.prenorm_partition_spec)
             x = self.norm(inputs)
+            x = maybe_shard(x, cfg.premlp_partition_spec)
```

Comment:
Do we need this? I suppose `norm` usually does not change the partition spec?

Reply:
We want to change the spec post norm to not propagate the spec from norm to MLP linear1. 
So for sequence parallel we would make the premlp_partition_spec = ((fsdp, data), None, None) to force an all-gather on the sequence dimension.

Reply:
Thanks for the explanation. Maybe we can add `output_partition_spec` to RMSNorm.Config to be consistent with Linear.Config. This is also more flexible as RMSNorm can be used in other places. WDYT?

Reply:
Yes, I actually originally had this! I have revised to take this approach.